### ISSUE TITLE
parseCoins has to work with Big Int arithmetics. Uint64 is not enough

### ISSUE THREAD

**Issue by AKorpusenko** [cosmos/cosmjs#1268]
`parseCoins` can work only with uint64 type which is too small for tokens with decimals like 12 or 18
https://github.com/cosmos/cosmjs/blob/main/packages/proto-signing/src/coins.ts#L20
has to be changed to work with BigInt and serialized to a string type after

**Comment by webmaster128** [comment]
Actually the only thing this line does is a format check on the string and strip leading zeros. So we can even implement it without using BigInt, which is [still an issue for some environments](https://github.com/cosmos/cosmjs/issues/1144).

**Comment by AKorpusenko** [comment]
@webmaster128, thanks for reply
But wtf you already use the big number arithmetics for Uint64 😄. 
[Here](https://github.com/cosmos/cosmjs/blob/main/packages/math/src/integers.ts#L2) there is a case of usage of `bn.js`. Moreover you do use the BigInt [here](https://github.com/cosmos/cosmjs/blob/main/packages/math/src/integers.ts#L261) 

The easiest case is just to use either the BigInt, or bn.js is you want so. Like this:

```ts
/* eslint-disable no-bitwise */
import BN from "bn.js";
import { Coin } from "@cosmjs/amino";
/**
 * Takes a coins list like "819966000ucosm,700000000ustake" and parses it.
 *
 * This is a Stargate ready version of parseCoins from @cosmjs/amino.
 * It supports more denoms.
 */
export function parseCoins(input: string): Coin[] {
  return input
    .replace(/\s/g, "")
    .split(",")
    .filter(Boolean)
    .map((part) => {
      // Denom regex from Stargate (https://github.com/cosmos/cosmos-sdk/blob/v0.42.7/types/coin.go#L599-L601)
      const match = part.match(/^([0-9]+)([a-zA-Z][a-zA-Z0-9/]{2,127})$/);
      if (!match) throw new Error("Got an invalid coin string");
      return {
        amount: new BN((match[1]), 10, "be").toString(),
        denom: match[2],
      };
    });
}
```

**Comment by webmaster128** [comment]
The point I am trying to make is that `parseCoins` does not do any arithmetics. It splits and checks a string and converts a string number to a string number after this has been checked to be `[0-9]+`. So we can get rid of the UintXYZ limitation entirely.

We use bn.js for historic reasons ans I'd like to avoid creating more usages of it. I'd like to use native BigInt but the React Native folks make me wonder if we can use it in critical places.

Anyways, I'll fix this one way or another.

**Comment by webmaster128** [comment]
Shipped as part of CosmJS 0.29.0
